components/src/preact/lineageFilter/fetchLineageAutocompleteList.spec.ts

```diff
@@ -4,25 +4,283 @@ import { fetchLineageAutocompleteList } from './fetchLineageAutocompleteList';
+                        [lineageField]: 'A',
+                        count: 1,
+                    },
+                    {
+                        [lineageField]: 'A.1',
+                        count: 2,
+                    },
+                    {
+                        [lineageField]: 'A.2',
+                        count: 3,
+                    },
+                    {
+                        [lineageField]: 'XA',
+                        count: 4,
+                    },
+                ],
+            },
+        );
+
+        lapisRequestMocks.lineageDefinition(
+            {
+                A: {
+                    aliases: ['a'],
+                },
+                'A.1': {
+                    parents: ['A'],
+                    aliases: ['a.1'],
+                },
+                'A.2': {
+                    parents: ['A'],
+                    aliases: ['a.1'],
+                },
+                XA: {
+                    aliases: ['xa'],
+                    parents: ['A.1', 'A.2'],
+                },
+            },
+            lineageField,
+        );
+
+        const result = await fetchLineageAutocompleteList({
+            lapisUrl: DUMMY_LAPIS_URL,
+            lapisField: lineageField,
+            lapisFilter,
+        });
+
+        expect(result).to.deep.equal([
+            {
+                lineage: 'A',
+                count: 1,
+            },
+            {
+                lineage: 'A*',
+                count: 10,
+            },
+
+            {
+                lineage: 'A.1',
+                count: 2,
+            },
+            {
+                lineage: 'A.1*',
+                count: 6,
+            },
+            {
+                lineage: 'A.2',
+                count: 3,
+            },
+            {
+                lineage: 'A.2*',
+                count: 7,
+            },
+            {
+                lineage: 'XA',
+                count: 4,
+            },
+            {
+                lineage: 'XA*',
+                count: 4,
+            },
+        ]);
+    });
+
+    test('should work with grandchildren', async () => {
+        lapisRequestMocks.aggregated(
+            { fields: [lineageField], ...lapisFilter },
+            {
+                data: [
+                    {
+                        [lineageField]: 'A',
+                        count: 1,
+                    },
+                    {
+                        [lineageField]: 'A.1',
+                        count: 2,
+                    },
+                    {
+                        [lineageField]: 'A.1.1',
+                        count: 3,
+                    },
+                ],
+            },
+        );
+
+        lapisRequestMocks.lineageDefinition(
+            {
+                A: {
+                    aliases: ['a'],
+                },
+                'A.1': {
+                    parents: ['A'],
+                    aliases: ['a.1'],
+                },
```

Comment:
Let's mix up the order a little to make sure that we're indifferent to how the file was sorted.
```suggestion
                'A.1': {
                    parents: ['A'],
                    aliases: ['a.1'],
                },
                A: {
                    aliases: ['a'],
                },
```